src/service.js

```diff
@@ -109,7 +110,7 @@ async function startService ({ peerId, port, peerAnnounceAddr, awsClient, connec
       service.handle(protocol, async ({ connection: dial, stream }) => {
         try {
           const connection = new Connection(stream)
-          const canceled = new LRU({ max: 200 })
+          const canceled = cancelsPerPeer.get(dial.remotePeer.toString()) || new LRU({ max: 200 })
```

Comment:
Ah, I meant remove it from the listener and do `cancelsPerPeer.set` here if not exists